packages/react/use-direction/src/useDirection.tsx

```diff
@@ -8,8 +8,11 @@ export function useDirection(element: HTMLElement | null, directionProp?: Direct
   const rAFRef = React.useRef<number>(0);
 
   React.useEffect(() => {
-    if (element && directionProp === undefined) {
-      const computedStyle = getComputedStyle(element);
+    // We check inherited direction of parent instead of `element` so that computed styles are
+    // not overridden by dir attribute on element if inherited direction changes. The `dir`
+    // attribute should always sync with direction prop OR its inherited direction.
```

Comment:
Does this work better?

```suggestion
    // We check inherited direction of the parent instead of the `element` itself.
    // This is because we internally set the computed `dir` on the element and so we wouldn't be
    // able to react to changes to inherit direction. The `dir` attribute we set should always sync
    // with the direction prop OR its inherited direction.
```